### ISSUE TITLE
Long loading times on startup and channel load

### ISSUE THREAD

**Issue by Tampa** [mattermost/mattermost-mobile#7660]
#### Summary
Loading takes longer than in previous versions, especially while loading channel contents.

### Environment Information
  - Device Name: HTC One m9, Doogee S88+
  - OS Version: LineageOS Android 9, Android 10
  - Mattermost App Version: 2.9.1
  - Mattermost Server Version: 9.1.0

#### Steps to reproduce
Load a server with multiple channels and direct messages, observe loading times.

Stop app, open via push notification of unread message, observe almost a minute of loading on slower devices.

#### Expected behavior
Previous versions of the app had much faster loading, especially of channel content.

#### Observed behavior (that appears unintentional)
Loading of the app itself is reasonably fast with the recent server versions, but loading channels takes a very long time, especially if the content is complex and already read. Channels with a lot of unread content tend to load faster for some reason. Even returning from a direct message to the channel list is often very slow.

It is worst when the app has been dormant or stopped entirely. The initial loading tends to be the biggest hurdle for it. The server and phone are located within the same network over 2.4ghz wifi and the server isn't experiencing any overload or other bottlenecks. The phone is a slightly older model, but newer devices show a similar behavior albeit less extreme(the relation of app and channel load times are the same just overall faster)

This started a few versions ago. I noticed the overall layout and design changes in the app which went along with really slow loading. Over time this got a bit better, but it is definitely still sluggish even on powerful devices. I did check server logs for any major errors, but can't see any errors come in when the app starts loading.

Some history of this is in ticket #7001 Opening separate ticket to de-clutter as per @larkox request.

**Comment by amyblais** [comment]
Opened https://mattermost.atlassian.net/browse/MM-55470.

**Comment by vetash** [comment]
After several years, I am forced to admit - the client is absolutely unusable, my entire team has split into several factions, some have moved to Telegram, others to WhatsApp, and a few to Discord. The software that was supposed to bring people together has divided them. Reinstallations help, but it's not a solution; it's impossible to reinstall the software for everyone at once.

There are at least 7 issues logged on GitHub, some of which are also registered in Jira, but there is no information there either.  

#7098 
#7202 
#7322 
#7202 
#7030 
#6062 

and this.

Unfortunately, I don't have the skills to try to fix the current situation. Developers, please tell me how much money is needed to fix the situation? There is a huge community here, and I believe we are capable of crowdfunding to fix this issue.

The problem persists on all mobile clients, both on iOS and Android, regardless of the hardware.

Please, save this software, it's dying!

**Comment by ShiroDN** [comment]
Yes, mobile apps become almost unusable a few days after reinstalling them. We moved from Rocket.Chat last year because the server app was slow and kept crashing, and now we are on Mattermost, but the mobile apps are bad. Initially, we were even considering a paid plan, but with the app in this state, I'm not even sure if we'll remain on Mattermost...

**Comment by dalekirkwood** [comment]
Hi Devs - Firstly, Mattermost is great.....on Desktop & Web. We are using for a year and recommend it to alot of companies.

But the Android app is bad. It's slow to load and quite buggy, no matter the device.

It is really frustrating for my team, especially when travelling. We are considering switching because of it.

Have you considered a 'Lite' App, something that is very lightweight, basic functionality.etc.etc.

Most of us just need to send a message.

**Comment by robby-ac** [comment]
> But the Android app is bad. It's slow to load and quite buggy, no matter the device.
> Have you considered a 'Lite' App, something that is very lightweight, basic functionality.etc.etc.

Like the idea of a Lite app, but I think it would at least need to support images with previews, other files could just be download only
